### PR TITLE
[4.0][com_contenthistory] convert to prepared statement 

### PR DESCRIPTION
**Pull request by alikon**
### Summary of Changes

use prepared statement for SQL

### Testing Instructions
test com_contenthistory

### Expected result
work as before

### Actual result
N/A

### CODE REVIEW

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 384b579e6efcb89b24a3adf001f9ca6f58d4bf31<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27084">issues.joomla.org/tracker/joomla-cms/27084</a>.</sub>

File: administrator/components/com_contenthistory/Helper/ContenthistoryHelper.php [hunk 1]
```diff
@@ -207,13 +208,15 @@ public static function getLookupValue($lookup, $value)
 	{
 		$result = false;
 
-		if (isset($lookup->sourceColumn) && isset($lookup->targetTable) && isset($lookup->targetColumn)&& isset($lookup->displayColumn))
+		if (isset($lookup->sourceColumn) && isset($lookup->targetTable) && isset($lookup->targetColumn) && isset($lookup->displayColumn))
 		{
-			$db = Factory::getDbo();
+			$db    = Factory::getDbo();
+			$value = (int) $value;
```
Comment (Quy) [inline administrator/components/com_contenthistory/Helper/ContenthistoryHelper.php:214]: Per the docblock, `$value` is mixed so should we assume it to be integer all the time?

Comment (alikon) [reply administrator/components/com_contenthistory/Helper/ContenthistoryHelper.php:214]: in theory maybe, but in pratice should always be the PK (id) most of  the times 99%

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 15d4fe543501c471c447d9352cd47d38f23ffbed<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27084">issues.joomla.org/tracker/joomla-cms/27084</a>.</sub>

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 4b07257f25f90adfb72e10e5fb21702f3808e7df<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27084">issues.joomla.org/tracker/joomla-cms/27084</a>.</sub>

**Comment by HLeithner** [comment]
Thanks

**Pull request merged**